include/nbl/builtin/hlsl/device_capabilities_traits.hlsl

```diff
@@ -7,15 +7,44 @@
 #include <nbl/builtin/hlsl/member_test_macros.hlsl>
 
 #ifdef __HLSL_VERSION
+
+NBL_GENERATE_MEMBER_TESTER(shaderFloat64);
+NBL_GENERATE_MEMBER_TESTER(shaderDrawParameters);
+NBL_GENERATE_MEMBER_TESTER(subgroupArithmetic);
+NBL_GENERATE_MEMBER_TESTER(fragmentShaderPixelInterlock);
+NBL_GENERATE_MEMBER_TESTER(maxOptimallyResidentWorkgroupInvocations);
+
+#define NBL_GENERATE_GET_OR_DEFAULT(field, ty, default) \
+template<typename S, bool = has_member_##field<S>::value> struct get_or_default_##field : integral_constant<ty,S::field> {}; \
+template<typename S> struct get_or_default_##field<S,false> : integral_constant<ty,default> {};
+
 namespace nbl
 {
 namespace hlsl
 {
+
+namespace impl
+{
+NBL_GENERATE_GET_OR_DEFAULT(shaderFloat64, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(shaderDrawParameters, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(subgroupArithmetic, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(fragmentShaderPixelInterlock, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(maxOptimallyResidentWorkgroupInvocations, uint16_t, 0);
+}
+
+
 template<typename device_capabilities>
 struct device_capabilities_traits
```

Comment:
cool now use it in the `nbl/builtin/hlsl/workgroup/arithmetic.hlsl` and let me know if ex 48 works